internal/config/config.go

```diff
@@ -638,6 +639,148 @@ func (c *Config) Validate(force bool) error {
+	// description: |
+	//   Lowest acceptable bootloader version.
+	BootloaderVersion uint8 `json:"bootloaderVersion" yaml:"bootloaderVersion"`
+	// description: |
+	//   Lowest acceptable TEE version.
+	TEEVersion uint8 `json:"teeVersion" yaml:"teeVersion"`
+	// description: |
+	//   Lowest acceptable SEV-SNP version.
+	SNPVersion uint8 `json:"snpVersion" yaml:"snpVersion"`
+	// description: |
+	//   Lowest acceptable microcode version.
+	MicrocodeVersion uint8 `json:"microcodeVersion" yaml:"microcodeVersion"`
+	// description: |
+	//   Configuration for validating the firmware signature.
+	FirmwareSignerConfig SNPFirmwareSignerConfig `json:"firmwareSignerConfig" yaml:"firmwareSignerConfig"`
+	// description: |
+	//   AMD Root Key certificate used to verify the SEV-SNP certificate chain.
+	AMDRootKey Certificate `json:"amdRootKey" yaml:"amdRootKey"`
+}
+
+// DefaultForAzureSEVSNP returns the default configuration for Azure SEV-SNP attestation.
+// TODO(AB#3042): replace with dynamic lookup for configurable values.
+func DefaultForAzureSEVSNP() AzureSEVSNP {
+	return AzureSEVSNP{
+		Measurements:      measurements.DefaultsFor(cloudprovider.Azure),
+		BootloaderVersion: bootloaderVersion,
+		TEEVersion:        teeVersion,
+		SNPVersion:        snpVersion,
+		MicrocodeVersion:  microcodeVersion,
+		FirmwareSignerConfig: SNPFirmwareSignerConfig{
+			AcceptedKeyDigests: idkeydigest.DefaultList(),
+			EnforcementPolicy:  idkeydigest.MAAFallback,
+		},
+		AMDRootKey: mustParsePEM(arkPEM),
+	}
+}
+
+// GetVariant returns azure-sev-snp as the variant.
+func (AzureSEVSNP) GetVariant() variant.Variant {
+	return variant.AzureSEVSNP{}
+}
+
+// GetMeasurements returns the measurements used for attestation.
+func (c AzureSEVSNP) GetMeasurements() measurements.M {
+	return c.Measurements
+}
+
+// SNPFirmwareSignerConfig is the configuration for validating the firmware signer.
+type SNPFirmwareSignerConfig struct {
+	// description: |
+	//   List of accepted values for the firmware signing key digest.\nValues are enforced according to the 'enforcementPolicy'\n     - 'equal'       : Error if the reported signing key digest does not match any of the values in 'acceptedKeyDigests'\n     - 'maaFallback' : Use 'equal' checking for validation, but fallback to using Microsoft Azure Attestation (MAA) for validation if the reported digest does not match any of the values in 'acceptedKeyDigests'. See the Azure docs for more details: https://learn.microsoft.com/en-us/azure/attestation/overview#amd-sev-snp-attestation\n     - 'warnOnly'    : Same as 'equal', but only prints a warning instead of returning an error if no match is found
+	AcceptedKeyDigests idkeydigest.List `json:"acceptedKeyDigests" yaml:"acceptedKeyDigests"`
+	// description: |
+	//   Key digest enforcement policy. One of {'equal', 'maaFallback', 'warnOnly'}
+	EnforcementPolicy idkeydigest.Enforcement `json:"enforcementPolicy" yaml:"enforcementPolicy" validate:"required"`
```

Comment:
should the enforcementPolicy explanations be moved to the description of EnforcementPolicy?

Reply:
With how the docgen tool works, setting the explanations on `EnforcementPolicy` results in the doc comment being behind, and under the entry in the config.
Unless there is an option to explicitly move the doc comment above the `EnforcementPolicy` key, I would keep it as is.

How I would like it to look:
```yaml
# ...
someKey: someValue # Some doc comment explaining the value
# First line of the enforcementPolicy doc comment
# Second line of the enforcementPolicy doc comment
# Third line of the enforcementPolicy doc comment
enforcementPolicy: MaaFallback
anotherKey: "anotherValue" # Another doc comment explaining a value
# ...
```

How it actually looks:
```yaml
# ...
someKey: someValue # Some doc comment explaining the value
enforcementPolicy: MaaFallback # First line of the enforcementPolicy doc comment
# Second line of the enforcementPolicy doc comment
# Third line of the enforcementPolicy doc comment
anotherKey: "anotherValue" # Another doc comment explaining a value
# ...
```

Reply:
I agree. I expected it would result in what you would like it to look. (It actually looks like this for acceptedKeyDigests, maybe because of its type. But I guess you know that already.)